### DISCUSSION TITLE
What is the correct / best workflow to duplicate an example to be able to edit parameters ?

### DISCUSSION THREAD

**Discussion by s-celles** [OpenModelica/OpenModelica#11221] [Q&A]
Hello,

I'm looking for a correct / best workflow to duplicate an example to be able to edit parameters.

Here is what I did

![image](https://github.com/OpenModelica/OpenModelica/assets/109167/85553a9e-8179-45bc-971c-2046809e99e8)

Contextual menu on an example (Modelica.Electrical.PowerConverters.Examples.DCDC.ChopperStepDown here) / Duplicate

![image](https://github.com/OpenModelica/OpenModelica/assets/109167/7581de30-354c-4fc1-b674-c1d44fb033b6)

![image](https://github.com/OpenModelica/OpenModelica/assets/109167/3af59d6d-a568-45fe-881d-b2e2926aa91a)

A lot seems broken.

I need to edit myself Modelica code, recreate connections...

Isn't there a better workflow ?

Kind regards

**Comment by adeas31** [comment]
If you are only looking to edit parameters then create a new model and extend it from the example.

The duplicate functionality is not fully developed yet and there are cases when it doesn't work properly. See https://github.com/OpenModelica/OpenModelica/issues/5346

**Reply by s-celles** [reply]
By extending you mean copy from example / pasting in a new class ?
I think most of the problem is probably because I don't see the difference between Model and Class with Modelica.

**Reply by adeas31** [reply]
> By extending you mean copy from example / pasting in a new class ?

No, when you create a new model then select the example in the `Extends` text field. This will make your model inherited by example and show all its components whose parameters you can edit.

**Reply by casella** [reply]
Extending means defining a new model (or class, they are synonims in this context) by inheritance. If you have a model `A`, you can define a new model `B`:
```modelica
model B
  extends A;
end B;
```
This is initially the same as `A`. But now, if `A` has some component `comp` with a parameter `p = 1`, you can open `B`, double-click on the component `comp` and change the value of the parameter. This will end up with the following code
```modelica
model B
  extends A(comp(p = 4));
end B;
```
Another thing you can do on an extended model is to add more stuff, e.g. drag in other components and connect them. What you can't do is remove components.

**Comment by casella** [comment]
Please check #5346, from the next nightly build this should be implemented. Please test it and report any issue in that ticket.
